gnmi/gnmit/gnmit.go

```diff
@@ -366,20 +346,53 @@ func set(schema *ytypes.Schema, cache *cache.Cache, target string, req *gpb.SetR
 // Set is a prototype for a gNMI Set operation.
 // TODO(wenbli): Add unit test.
 func (s *GNMIServer) Set(ctx context.Context, req *gpb.SetRequest) (*gpb.SetResponse, error) {
-	s.c.smu.Lock()
-	defer s.c.smu.Unlock()
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
```

Comment:
we shouldn't require external users (ondatra tests) to have to set some metadata for set requests to work

Reply:
It by default uses ConfigMode, added comment to make that clearer.

Reply:
this still requires metadata to be set, which is always true

Reply:
Done.